### PR TITLE
crypto: migrate to SPDX identifier

### PR DESCRIPTION
**Pull request by jerpelea**
## Summary
Most tools used for compliance and SBOM generation use SPDX identifiers This change brings us a step closer to an easy SBOM generation.

Please note that there are licenses that have no SPDX identification

## Impact
SBOM

## Testing
CI

### CODE REVIEW

**Comment by jerpelea** [comment]
@xiaoxiang781216 
please help identify the license for the files identified with ???

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 please help identify the license for the files identified with ???

@ThePassionate please take a look.

**Comment by jerpelea** [comment]
@xiaoxiang781216 please provide the information

File: crypto/md5.c [hunk 1]
```diff
@@ -1,5 +1,8 @@
 /****************************************************************************
  * crypto/md5.c
+ *
+ * SPDX-License-Identifier: ???
```
[the remark this answers is not in the document]
Comment (ThePassionate) [reply crypto/md5.c:4]: 'Public Domain'

File: crypto/rijndael.c [hunk 2]
```diff
@@ -1,5 +1,8 @@
 /****************************************************************************
  * crypto/rijndael.c
+ *
+ * SPDX-License-Identifier: ???
```
Comment (ThePassionate) [inline crypto/rijndael.c:4]: 'Public Domain'

File: crypto/cast.c [hunk 3]
```diff
@@ -1,5 +1,8 @@
 /****************************************************************************
  * crypto/cast.c
+ *
+ * SPDX-License-Identifier: ???
```
[the remark this answers is not in the document]
Comment (jerpelea) [reply crypto/cast.c:4]: please specify the line

File: crypto/md5.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (jerpelea) [reply crypto/md5.c:4]: we ca not use public domain code

File: crypto/poly1305.c [hunk 4]
```diff
@@ -1,5 +1,8 @@
 /****************************************************************************
  * crypto/poly1305.c
+ *
+ * SPDX-License-Identifier: ???
```
[the remark this answers is not in the document]
Comment (jerpelea) [reply crypto/poly1305.c:4]: concluded MIT

File: crypto/rijndael.c (hunk 2 above)
Comment (jerpelea) [reply crypto/rijndael.c:4]: we ca not use PD

File: crypto/sha1.c [hunk 5]
```diff
@@ -1,5 +1,8 @@
 /****************************************************************************
  * crypto/sha1.c
+ *
+ * SPDX-License-Identifier: ???
```
[the remark this answers is not in the document]
Comment (jerpelea) [reply crypto/sha1.c:4]: please specify the line

File: crypto/xform.c [hunk 6]
```diff
@@ -1,5 +1,8 @@
 /****************************************************************************
  * crypto/xform.c
+ *
+ * SPDX-License-Identifier: ???
```
[the remark this answers is not in the document]
Comment (jerpelea) [reply crypto/xform.c:4]: proof?

File: crypto/castsb.h [hunk 7]
```diff
@@ -1,5 +1,8 @@
 /****************************************************************************
  * crypto/castsb.h
+ *
+ * SPDX-License-Identifier: ???
```
[the remark this answers is not in the document]
Comment (jerpelea) [reply crypto/castsb.h:4]: please specify the line

File: crypto/md5.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (raiden00pl) [reply crypto/md5.c:4]: @jerpelea why not? what about this: https://www.apache.org/legal/resolved.html#handling-public-domain-licensed-works ?

[the remark this answers is not in the document]
Comment (jerpelea) [reply crypto/md5.c:4]: @raiden00pl I can not see a Creative Commons license . The public domain is handled in a different way in EU and US

File: crypto/rijndael.c (hunk 2 above)
Comment (jerpelea) [reply crypto/rijndael.c:4]: The public domain is handled in a different way in EU and US

File: crypto/md5.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (raiden00pl) [reply crypto/md5.c:4]: @jerpelea what about this:

```
A work should be treated as being in the public domain when one of the following applies:
1. the work is covered by
   - the Creative Commons Public Domain Mark
   - a suitable dedication (to the public domain) by the authors
```
Are these conditions `OR` or `AND` ? It's not clear at all.

**Comment by jerpelea** [comment]
I removed all unclear licenses. Please merge the PR and I will start discussion on dev mailing list

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by cederom — APPROVED** [review]
Thank you folks great job! :-)

Btw isn't OpenBSD a BSD license? :-) :-)

**Comment by jerpelea** [comment]
> Thank you folks great job! :-)
> 
> Btw isn't OpenBSD a BSD license? :-) :-)

the license is BSD when the license is stated BSD

**Pull request merged**